main/lsp/LSPTypechecker.cc

```diff
@@ -642,6 +641,55 @@ LSPTypechecker::SlowPathResult LSPTypechecker::runSlowPath(LSPFileUpdates &updat
     }
 }
 
+void LSPTypechecker::cacheOpenFiles(absl::Span<const ast::ParsedFile> indexed, std::vector<core::FileRef> openFiles,
+                                    WorkerPool &workers) {
```

Comment:
I'm not 100% sold on this change. I think it might be better to clean out the cache of indexed trees completely, and wait for the fast path to repopulate it.